### PR TITLE
PR for Filters on Products

### PR DESCRIPTION
**Pull request by manojasarsa**
This PR includes:
- Category FIlter
- Price Range Filter
- Sort By Price Filter
- Ratings Filter
- Include Out of Stock Filter
- Fast Delivery Filter
- Product Listing

### CODE REVIEW

File: src/contexts/filterContext.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { createContext, useContext, useReducer } from "react";
+import { filterReducer } from "../reducer/filterReducer";
+
+const filterContext = createContext();
+
+const FilterProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(filterReducer, {
+    sortBy: null,
+    categories: {men: false, women: false, boys: false, girls: false},
```
Comment (mimansha-swarup) [inline src/contexts/filterContext.js:9]: nice approach

**Review by mimansha-swarup — APPROVED** [review]
code looks clean 
good to proceed

File: src/components/ProductCard/ProductCard.jsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+const ProductCard = ({product}) => {
+
+    console.log(product);
+
+    const {title, description, image, price, discountedPrice } = product;
+    // rating,
+        // inStock,
+        // fastDelivery,
+
```
Comment (gopalbbharadva) [inline src/components/ProductCard/ProductCard.jsx:6]: remove it if it's not intentional.

**Review by gopalbbharadva — APPROVED** [review]
Nice code structure, good to go. I learnt that we can give title of product in "alt" attribute of img.

File: src/components/ProductCard/ProductCard.jsx (hunk 2 above)
Comment (manojasarsa) [reply src/components/ProductCard/ProductCard.jsx:6]: Thanks for pointing out, Gopal. 
I'll do that.

**Pull request merged**